src/OrleansProviders/Streams/Memory/MemoryBatchContainer.cs

```diff
@@ -9,10 +7,10 @@
 
 namespace Orleans.Providers
 {
-    [Serializable]
     internal class MemoryBatchContainer<TSerializer> : IBatchContainer
-        where TSerializer : IMemoryMessageBodySerializer, new()
+        where TSerializer : IMemoryMessageBodySerializer
     {
+        private readonly IMemoryMessageBodySerializer serializer;
```

Comment:
you dropped the  [NonSerialized] attribute. Do you need to add it back or was it by design? Would this cause upgrade issues?